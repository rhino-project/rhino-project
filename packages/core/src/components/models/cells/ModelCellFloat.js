import { useGlobalComponent } from 'rhino/hooks/overrides';
import CellFloat from 'rhino/components/table/cells/CellFloat';

export const ModelCellFloatBase = (props) => <CellFloat {...props} />;

const ModelCellFloat = (props) =>
  useGlobalComponent('ModelCellFloat', ModelCellFloatBase, props);

export default ModelCellFloat;
