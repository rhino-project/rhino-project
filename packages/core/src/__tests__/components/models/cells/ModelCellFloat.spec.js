import ModelCellFloat from 'rhino/components/models/cells/ModelCellFloat';
import { sharedCellTests } from './sharedCellTests';

describe('ModelCellFloat', () => {
  sharedCellTests(ModelCellFloat);
});
