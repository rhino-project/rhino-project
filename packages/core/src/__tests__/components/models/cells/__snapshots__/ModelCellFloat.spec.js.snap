// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCellFloat should render the component with className from inherited props 1`] = `
<DocumentFragment>
  <div
    class="dummy-class"
  >
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellFloat should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`ModelCellFloat should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellFloat should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellFloat should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellFloat should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
